src/borg/archiver.py

```diff
@@ -1220,6 +1220,8 @@ def build_parser(self, prog=None):
         subparser.set_defaults(func=self.do_serve)
         subparser.add_argument('--restrict-to-path', dest='restrict_to_paths', action='append',
                                metavar='PATH', help='restrict repository access to PATH')
+        subparser.add_argument('--append-only', dest='append_only', action='store_true',
+                               help='only allow appending to files')
```

Comment:
Maybe just say "enable append-only mode regardless of repository config". That thing is a bit too complex to describe in a --help synopsis imho.